### DISCUSSION TITLE
[Help] Authorization Configuration

### DISCUSSION THREAD

**Discussion by henjoe** [nanomq/nanomq#1118] [General]
Hi, I want to only authorize "hmd" user to publish and subscribe to nanomq and everything is denied. 

I have below my config file, but seems like it doesn't apply after I restarted my nanomq broker.
 Am I doing wrong with my config?

Thanks.
```
authorization {
	no_match = allow
	deny_action = disconnect

	cache = {
		enable = true
		max_size = 32
		ttl = 1m
	}

	sources = [
		{
			type = simple
			enable = false

			users = [
				{
					username = "hmd"
					password = "hmd"
				}
				{
					username = "client"
					password = "public"
				}
			]
		},
		{
			type = file
			enable = true

			rules = [
				## Allow MQTT client using username "dashboard"  to subscribe to "$SYS/#" topics
				{"permit": "allow", "username": "dashboard", "action": "subscribe", "topics": ["$SYS/#"]}

				## Deny "All Users" subscribe to "$SYS/#" "#" Topics
				{"permit": "deny", "username": "#", "action": "subscribe", "topics": ["$SYS/#", "#"]}

				## Allow any other publish/subscribe operation
				##{"permit": "allow"}
				{"permit": "allow", "username": "hmd", "action": "pubsub", "topics": ["$SYS/#", "#"]}
				
				{"permit": "deny"}
			]
		}

```

**Comment by JaylinYu** [comment]
disable
allow_anonymous = false
then
nanomq start --conf /path_to_your_config

**Comment by henjoe** [comment]
Thanks @JaylinYu 

I already set the allow_anonymous to false and do the 
```
nanomq.exe start --config "path_of_my_custom_config"

```
![image](https://user-images.githubusercontent.com/49782655/227751130-6878767e-cb15-433c-8337-c17a6617289a.png)

Using nodered, I can still connect to nanomq broker even without using a credentials (without username and password).

**Comment by JaylinYu** [comment]
Could you also upload DEBUG level log file of nanomq? Perhaps your config file does not take effect
